### ISSUE TITLE
whitespace around operators in function keyword arguments

### ISSUE THREAD

**Issue by htyeim** [julia-vscode/DocumentFormat.jl#102]
I'm not sure if it is a problem. I think the spaces surrounding function arguments are usually unnecessary.

But the following code shows that `DocumentFormat.jl` will add spaces around arguments:
```
a = """
function test_format(a=1, b=2)
    a=1
end
""" 

print(DocumentFormat.format(a))

function test_format(a = 1, b = 2)
    a = 1
end
```
So it will affect autoformating in vs-code whether `julia.format.ops` is true or not. I don't if it is intend to act as this or not.

Some references:

https://github.com/julia-vscode/julia-vscode/issues/425

https://github.com/julia-vscode/DocumentFormat.jl/commit/b34757eae00ef8904b7dfd447b28b17b64a02d2b

**Comment by davidanthoff** [comment]
Agreed, the spaces in args seem weird to me as well.

What is the convention for this in Julia base?

**Comment by htyeim** [comment]
In `julia/base`, I found different ways:

without spaces: https://github.com/JuliaLang/julia/blob/master/base/io.jl#L1126

with spaces: https://github.com/JuliaLang/julia/blob/master/base/strings/search.jl#L28

with spaces: https://github.com/JuliaLang/julia/blob/master/base/version.jl#L59

This may be due to some historic reasons ... So I am not sure what is exactly the convention.

**Comment by non-Jedi** [comment]
For what it's worth, both the [Blue Style Guide](https://github.com/invenia/BlueStyle#whitespace) and [YASGuide](https://github.com/jrevels/yasguide#linealignmentspacing-guidelines) call for spaces around operators in general but carve out an exception for the `=` in keyword arguments and default arguments. By my reading [the JuMP style guide](http://www.juliaopt.org/JuMP.jl/v0.21.1/style/) does not carve out an exception for this case. [JuliaFormatter.jl is configurable in this](https://domluna.github.io/JuliaFormatter.jl/stable/api/#JuliaFormatter.format_text-Tuple%7BAbstractString%7D) but defaults to having whitespace.

On a personal level, because of my previous familiarity with [python's PEP8](https://www.python.org/dev/peps/pep-0008/#other-recommendations), no spaces looks far more natural to me.

In summary:

|   | Whitespace around `=` in keyword arguments? | Configurable |
| - | - | - |
| Julia Base | mixed | |
| Blue Style Guide | no | |
| YASGuide | no | |
| JuMP Guide | yes | |
| JuliaFormatter.jl | yes | yes |
| PEP8 | no | |

**Comment by ZacLN** [comment]
I feel like we used to have it the other way around and we changed it following a few requests -I prefer it without spaces myself. Anyway, it's easily configurable - does anyone feel up to a PR? The relevant code is at https://github.com/julia-vscode/DocumentFormat.jl/blob/master/src/passes.jl#L88-L90

**Comment by davidanthoff** [comment]
I'm fine with adding an option, but should we maybe change the default to not use whitespaces? I do feel that our default choices are quite key here :)

**Comment by ZacLN** [comment]
I'm happy with that!

**Comment by atbug** [comment]
The current description of the option seems to be the opposite of what is implemented at this moment.
![image](https://user-images.githubusercontent.com/8522036/84178080-df152800-aa51-11ea-83c5-75d210fc4efc.png)

**Comment by ZacLN** [comment]
Would the preference here be to enforce no spaces by default, with the other options being to enforce a space, and to do nothing?

**Comment by davidanthoff** [comment]
Yeah, an enum with three options seems the right thing: "do nothing", "enforce no whitespace" and "enforce one whitespace", right?

**Comment by atbug** [comment]
Should NamedTuple follow the same rule as keyword arguments?

**Comment by davidanthoff** [comment]
Could you open a new issue for that?
